lib/ruboty/actions/jobcan.rb

```diff
@@ -23,48 +23,101 @@ def remember_group_id
         message.reply("I remember.")
       end
 
-      def punch_clock(in_out = :auto)
-        unless (code = user_data["code"])
+      def register_group_alias
+        user_data["alias_#{message[:group_name]}"] = message[:group_id]
+        message.reply("I registered alias #{message[:group_name]} to ID #{message[:group_id]}.")
+      end
+
+      def punch_clock(in_out = :auto, at: nil)
+        unless (code = user_data["code"]) || login_env_present?
           message.reply("I don't know your JOBCAN code.")
           return
         end
-        unless (group_id = user_data["group_id"])
-          message.reply("I don't know your JOBCAN group ID.")
-          return
-        end
+
+        group_id = if at
+                     user_data["alias_#{at}"] or
+                       return cannot_find_alias_message(at)
+                   elsif !user_data["group_id"].nil?
+                     user_data["group_id"]
+                   else
+                     user_data["group_id"] or
+                       return message.reply("I don't know your JOBCAN group ID.")
+                   end
```

Comment:
The elsif block and the else block do the same thing -- both tries to fetch `user_data["group_id"]`. The elsif block could be removed.